x-pack/plugins/snapshot_restore/server/lib/index.ts

```diff
@@ -3,4 +3,6 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+
 export { booleanizeSettings } from './booleanize_settings';
+export { deserializeSnapshotSummary, deserializeSnapshotDetails } from './snapshot_serialization';
```

Comment:
I left out serialization/deserialization for the repository APIs, partly because there are 7 different repository types, each with different settings, so would result in either complex logic or a pair of methods for each type! but I also found that I haven't gotten much value out of performing deserialization, just that keys are camel cased. how do you feel about it?

Reply:
> I left out serialization/deserialization for the repository APIs, partly because there are 7 different repository types, each with different settings, so would result in either complex logic or a pair of methods for each type!

I haven't looked closely but that description sounds pretty arduous! As a consumer I think I'd generally prefer consistency but I am OK applying de/serialization on a case-by-case basis and skipping it for repositories for now. If consuming snake-cased strings doesn't impede writing and reading the code on the front-end then I don't see a problem with it.

I implemented it here at Seb's request because it was a trivial change to make. Also, as I write this I realize that the `SnapshotSummary` type excludes the `id` and `repository` properties that are on the `SnapshotSummaryEs` type because there is some data munging going on. So in this case, de/serialization does play a role other than just converting name patterns.

Reply:
I understand what you're saying Jen, but having a serialization has several benefits to me:

- we have a **single place** where we control what is sent to the client from a raw ES resource. This means that is at any moment we have a requirement and need to add a prop (a formatted date for example), we do it there and know that the client will have it (coming from a list of resource, a resource updated or a single resource GET request).
- consistency 😊 I love consistency and it feels strange to have different name cases across apps
- avoid overriding eslint rules each time we'd need to access a deconstructed prop